enzyme/Enzyme/GradientUtils.cpp

```diff
@@ -4956,7 +4956,7 @@ llvm::Value *GradientUtils::recursiveFAdd(llvm::IRBuilder<> &B,
     llvm::Type *rhs_ty = rhs->getType();
     for (auto idx : rhs_off)
       rhs_ty = getSubType(rhs_ty, idx);
-    assert(lhs_ty == rhs_ty);
+    //assert(lhs_ty == rhs_ty);
```

Comment:
Is this needed ?

Reply:
I assert here in the Julia specialfunctions test set on complex input. Will dig into it but nothing broke without it.

Reply:
We sometimes have {double, double} vs [2 x double]. I've created a helper that turns the array into an equivalent struct for the purposes of the assert.